packages/react-components/src/components/DetailsCard/DetailsCard.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+
+import {
+  ChevronRight,
+  ChevronDown,
+} from '@livechat/design-system-icons/react/tabler';
+import cx from 'clsx';
+
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { Heading } from '../Typography';
+
+import styles from './DetailsCard.module.scss';
+
+export interface IDetailsCardProps {
+  /**
+   * The CSS class for accordion container
+   */
+  className?: string;
+  /**
+   * Set the label
+   */
+  label: string;
+  /**
+   * Define if divider should be visible
+   */
+  withDivider?: boolean;
+  /**
+   * Removes the spacing inside the main container
+   */
+  fullSpaceContent?: boolean;
+  /**
+   * Set if card should be open by default
+   */
+  openOnInit?: boolean;
+  /**
+   * Set to hide the label on card open
+   */
+  hideLabelOnOpen?: boolean;
+}
+
+const baseClass = 'details-card';
+
+export const DetailsCard: React.FC<IDetailsCardProps> = ({
+  children,
+  className,
+  label,
+  withDivider,
+  fullSpaceContent,
+  openOnInit = false,
+  hideLabelOnOpen,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(openOnInit);
+  const mergedClassNames = cx(
+    styles[baseClass],
+    withDivider && styles[`${baseClass}--with-divider`],
+    className
+  );
+
+  return (
+    <div className={mergedClassNames}>
+      <button
+        className={cx(
+          styles[`${baseClass}__button`],
+          hideLabelOnOpen && isOpen && styles[`${baseClass}__button--hide`]
+        )}
+        onClick={() => setIsOpen(!isOpen)}
```

Comment:
While operating on the existing state we should use an updated function, so please create a separate handler `handleHideButtonClick`, something like this:
```
const handleHideButtonClick = () => setIsOpen(prevValue => !prevValue);
```